### PR TITLE
issue-26: Implemented RSA

### PR DESCRIPTION
**Pull request by toniukan**

### CODE REVIEW

File: src/core/cryptography/test/RSATest.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        return res
+    }
+
+    @Test fun fixedKeyRandomString(): Unit {
+        val random = SecureRandom()
+
+        val rsa = RSA(1024)
+        for (i in 1..10) {
+            val message = randomString(random.nextInt(20) + 1, 10, 256, random)
```
Comment (Fly-Style) [inline src/core/cryptography/test/RSATest.kt]: Please, don`t use magic numbers in functions signature, it`s really makes code harder to understand

Comment (Fly-Style) [inline src/core/cryptography/test/RSATest.kt]: Please, don`t use magic numbers in functions signature, it`s really makes code harder to understand.

File: src/core/cryptography/test/RSATest.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            val encrypted = rsa.encrypt(message)
+            assertEquals(message, rsa.decrypt(encrypted))
+        }
+    }
+
+    @Test fun randomKeyFixedString(): Unit {
+        val random = SecureRandom()
+        val message = randomString(random.nextInt(20) + 1, 10, 256, random)
```
Comment (Fly-Style) [inline src/core/cryptography/test/RSATest.kt]: Same problem.

File: src/core/cryptography/main/RSA.kt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    /**
+     * Generate a new public and private key set.
+     */
+    fun generateKeys() {
+        val r = SecureRandom()
+        val p = BigInteger(bitlen / 2, 100, r)
+        val q = BigInteger(bitlen / 2, 100, r)
+        n = p.multiply(q)
+        val m = p.subtract(BigInteger.ONE).multiply(q
```
Comment (Fly-Style) [inline src/core/cryptography/main/RSA.kt]: Change the chain location in this piece of code. I has written you how you can make it better.

**Comment by toniukan** [comment]
Updated. Please check.

**Pull request merged**